### ISSUE TITLE
Angular InstantSearch failing to compile when using node version 14

### ISSUE THREAD

**Issue by SheLovesCode** [algolia/angular-instantsearch#967]
<!--
  Thanks for participating in this project! We will try to answer as soon as possible 🙏
-->

**Describe the bug 🐛**

We are currently trying to run Angular InstantSearch using one of your code samples repository [Doc Code Samples](https://github.com/algolia/doc-code-samples/) and the project is failing to comple after running `npm install .` using node 14. The project runs perfectly when node 16 is used.

**To Reproduce 🔍**


```
1. git clone https://github.com/algolia/doc-code-samples/
2. nvm use 14.16.1
3. node -v 
  This should show that node 14.16.1 is in use
5. cd doc-code-samples/angular-instantsearch/getting-started
6. npm install . && npm run start
```

**Expected behavior 💭**

The build fails to compile, throwing the following errors: 

****************************************************************************************************************************************************

Error: node_modules/@algolia/ui-components-highlight-vdom/dist/esm/Highlight.d.ts:1:23 - error TS2688: Cannot find type definition file for 'react'.

1 /// <reference types="react" />
                        ~~~~~

Error: node_modules/@algolia/ui-components-highlight-vdom/dist/esm/Highlight.d.ts:33:209 - error TS2503: Cannot find namespace 'JSX'.

33 export declare function createHighlightComponent({ createElement, Fragment, }: Renderer): ({ parts, highlightedTagName, nonHighlightedTagName, separator, className, classNames, ...props }: HighlightProps) => JSX.Element;
                                                                                                                                                                                                                   ~~~

*****************************************************************************************************************************************************

**Screenshots 🖥**

The complete error log can be seen below:

![Screenshot from 2022-10-03 15-17-56](https://user-images.githubusercontent.com/40703627/193589971-16cbe939-042f-48ba-8203-27e1c6c3eab4.png)

**Environment:**

 - OS: Ubuntu 20.04.1 LTS
 - Node Version: 14.16.1
 - Angular Version: 12.2.1
 - Algolia search version: 4.14.2
 - Algolia instant search version: 4.3.0

**Comment by Emily-RoseSteyn** [comment]
I managed to also reproduce this by running `npx create-instantsearch-app my-app --template "Angular InstantSearch"`. I was still getting this issue in Node 16 when using the template above. I was trying various solutions and the only thing that seemed to solve the issue is when I installed `@types/react` to my dev dependencies. This seems a bit weird considering it is an Angular app.

**Comment by Haroenv** [comment]
Do you also have this if you have the latest version of InstantSearch.js (underlying)? The fix should have been released in 4.46.2 https://github.com/algolia/instantsearch.js/issues/5109#issuecomment-1254653928

**Comment by Haroenv** [comment]
Thanks for that PR, it indeed will cause people to have minimally have this issue @Emily-RoseSteyn

**Comment by Haroenv** [comment]
Sorry for the delay here, this involved a lot of moving pieces. To ensure it's definitely updated, we've released 4.3.1 of Angular InstantSearch that fixes the issue
